src/module_2/analyzing_csv_file.md

```diff
@@ -0,0 +1,197 @@
+    count_children                      0
+    count_babies                        0
+    count_pets                          0
+    people_ex_baby                      0
+    days_since_purchase_variant_id      0
+    avg_days_to_buy_variant_id          0
+    std_days_to_buy_variant_id          0
+    days_since_purchase_product_type    0
+    avg_days_to_buy_product_type        0
+    std_days_to_buy_product_type        0
+    dtype: int64
+
+
+
+No row nulls in the data. 
+
+
+```python
+interesting_cols = ['outcome','global_popularity', 'normalised_price', 'set_as_regular', 'count_adults', 'count_children'
+                , 'count_babies', 'count_pets', 'ordered_before', 'abandoned_before']
+
+correlation_matrix = df[interesting_cols].corr(numeric_only=True)
+
+plt.figure(figsize=(10, 6))
+sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', fmt='.2f')
+plt.title('Correlation between Outcome and Other Features')
+plt.show()
+```
+
+
+    
+![png](analyzing_csv_file_files/analyzing_csv_file_5_0.png)
+    
+
+
+
+```python
+bought_products = df[df['outcome'] == 1]
+
+info_per_order = bought_products.groupby('order_id').size()
+info_per_order.describe()
+```
+
+
+
+
+    count    3427.000000
+    mean        9.697111
+    std         6.789204
+    min         1.000000
+    25%         5.000000
+    50%         9.000000
+    75%        13.000000
+    max        78.000000
+    dtype: float64
+
+
+
+
+```python
+product_outcome_means = df.groupby('product_type')['outcome'].mean().sort_values(ascending=False).head(10).reset_index()
+
+plt.figure(figsize=(12, 6))
+sns.barplot(x=product_outcome_means['product_type'], y=product_outcome_means['outcome'])
+plt.title('Top 10 Product Types by Outcome Mean (Buy Probability)')
+plt.xlabel('Product type')
+plt.ylabel('Outcome Mean')
+plt.xticks(rotation=45)
+plt.show()
+```
+
+
+    
+![png](analyzing_csv_file_files/analyzing_csv_file_7_0.png)
+    
+
+
+After calculating the 'Outcome' mean of each type_product, the number of times it is purchased with respect to the entire dataset, we can observe the probability of purchase for each type of product. With this information we can see that the products with the highest probability of purchase are staple products that everyone uses and that are purchased frequently, such as toilet rolls or bin bags.
+
+Now let's try to observe the evolution over time of the most sold types of products, to see if any of them have a different trend or follow a different pattern of purchase depending on the date.
+
+
+```python
+top_product_types = bought_products['product_type'].value_counts().head(5).index
+
+df_top_products = bought_products[bought_products['product_type'].isin(top_product_types)]
+df_top_products['order_date'] = df_top_products['order_date'].dt.to_period('M').dt.to_timestamp()
+df_grouped = df_top_products.groupby([df_top_products['order_date'], 'product_type']).size().reset_index(name='count')
+```
+
+
+```python
```

Comment:
As you say in the next section, the data for the last month is not complete. In this case, I would have just removed it from this plot to not give room for an error interpreting it.